src/tbx/mod.rs

```diff
@@ -151,16 +151,22 @@ impl Reader {
         let path = ffi::CString::new(path).unwrap();
         let c_str = ffi::CString::new("r").unwrap();
         let hts_file = unsafe { htslib::hts_open(path.as_ptr(), c_str.as_ptr()) };
-        unsafe {
-            if (*hts_file).format.category != htslib::htsFormatCategory_region_list
-                && (*hts_file).format.format != htslib::htsExactFormat_sam
-            {
-                return Err(Error::InvalidIndex);
-            }
+        let file_format = unsafe { (*hts_file).format.format };
+        let format_category = unsafe { (*hts_file).format.category };
+        match (format_category, file_format) {
+            (htslib::htsFormatCategory_region_list, _) => (),
+            (_, htslib::htsExactFormat_sam) => (),
+            // A workaround for htsLib's failure to correctly recognize the format of a
+            // simple BED file with a header line. See #194
+            (htslib::htsFormatCategory_unknown_category, htslib::htsExactFormat_text_format) => (),
+            _ => return Err(Error::InvalidIndex),
         }
 
         let hts_format = unsafe { (*htslib::hts_get_format(hts_file)).format };
         let tbx = unsafe { htslib::tbx_index_load(path.as_ptr()) };
+        if tbx.is_null() {
+            return Err(Error::InvalidIndex);
+        }
```

Comment:
The check for nullity is moved earlier to prevent a possible null dereference in the block below.

Reply:
Jep, moving this up here makes this safer and removing the above file type checks should be perfectly fine. Using `htslib`-functionality like `hts_get_format` should always be the preferred way of doing this and in the future, I guess we should try to document any cases where we deviate from this with a quick code comment stating why...